README.md

```diff
@@ -13,46 +13,133 @@ Features
+# 6. Start up the wagtaildemo development server.
+./manage.py runserver 0.0.0.0:8000
+# Success!
+```
+
+- Visit your site at http://localhost:8000
+- The admin interface is at http://localhost:8000/admin/ - log in with `admin` / `changeme`.
 
 What you can do
 ---------------
 
-Start up wagtaildemo:
-
-    ./manage.py runserver 0.0.0.0:8000
+> Note: all of those commands are meant to be used **inside the Vagrant virtual machine**. To get there, go to your local Wagtail (`cd ~/Development/wagtail`) set up and `vagrant up` then `vagrant ssh`.
 
-and visit http://localhost:8000. The admin interface is at http://localhost:8000/admin/ - log in with admin / changeme
+Start the wagtaildemo server:
 
-Run tests:
+```sh
+./manage.py runserver 0.0.0.0:8000
+# Then visit http://localhost:8000 in your browser.
+```
 
-    cd ../wagtail
-    ./runtests.py
+Run the tests:
 
-Build front-end assets:
+```sh
+cd /home/vagrant/wagtail
+# Python tests.
+./runtests.py
+# Node tests.
+npm run test
+```
 
-    cd ../wagtail
-    npm run build
+Run the linting:
 
-(or `npm run watch` to watch the source files for changes)
+```sh
+cd /home/vagrant/wagtail
+# Python linting.
+# TODO
+# JavaScript linting.
+npm run lint
+```
 
-Build the docs:
-
-    cd ../wagtail/docs
-    make html
-
-Switch to Python 2.7:
-
-    source ~/.virtualenvs/wagtailpy2/bin/activate
-
-And back to Python 3:
+Build front-end assets:
 
-    source ~/.virtualenvs/wagtaildemo/bin/activate
+```sh
+cd /home/vagrant/wagtail
+npm run build
+# Note: can also be done from outside the virtual machine, provided you have the right version of node installed.
+```
+
+Start front-end development tools and file watching:
+
+```sh
+cd /home/vagrant/wagtail
+npm run start
+# Note: can also be done from outside the virtual machine, provided you have the right version of node installed.
+```
+
+Build the documentation:
+
+```sh
+cd /home/vagrant/wagtail/docs
+make html
+```
+
+Switch between Python versions (Wagtail supports multiple ones):
+
+```sh
+# To switch to python 2.7,
+source ~/.virtualenvs/wagtailpy2/bin/activate
+# And to switch back to Python 3,
+source ~/.virtualenvs/wagtaildemo/bin/activate
+```
+
+Getting ready to contribute
+---------------------------
+
+Here are other actions you will likely need to do to make your first contribution to the project.
+
+Set up git remotes to Wagtail forks:
+
+```sh
+cd ~/Development/wagtail/wagtail
+# Change the default origin remote to point to your fork.
+git remote set-url origin https://github.com/<USERNAME>/wagtail.git
+# Add wagtail/wagtail as a "torchbox" remote.
+git remote add torchbox git@github.com:wagtail/wagtail.git
+# Add springload/wagtail as a "springload" remote.
+git remote add springload git@github.com:springload/wagtail.git
+# Add gasman/wagtail as a "gasman" remote.
+git remote add gasman git@github.com:gasman/wagtail.git
+# Pull latest changes from all remotes / forks.
+git pull --all
+```
+
+Get your development environment configured:
+
```

Comment:
Is this referring to IDEs? I think we're better leaving this out - advocating one editor or another really is opening a can of worms :-)